cmd/secrets-provider/main.go

```diff
@@ -35,7 +35,7 @@ func main() {
 	}
 
 	// verify we don't run with a K8s secrets storage configuration in a side car
-	if secretsConfig.StoreType == secretsConfigProvider.K8S && authnConfig.ContainerMode != "init" {
+	if secretsConfig.StoreType == secretsConfigProvider.K8S && (authnConfig.ContainerMode != "init" && authnConfig.ContainerMode != "application") {
```

Comment:
i think you need to rebase on master. please do so and ping me when the PR is ready.

Reply:
@orenbm we should be good to go

Reply:
Please, turn "init" and "application" to constant String parameters